make/targets/openshift/crd-schema-gen.mk

```
@@ -54,16 +61,58 @@ verify-codegen-crds-$(1): update-codegen-crds-$(1)
 verify-codegen-crds: verify-codegen-crds-$(1)
 .PHONY: verify-codegen-crds
 
+update-codegen-techpreview-crds-$(1): ensure-controller-gen ensure-yq ensure-yaml-patch
+	OPENSHIFT_REQUIRED_FEATURESET=TechPreviewNoUpgrade $(call run-crd-gen,$(2),$(3))
+.PHONY: update-codegen-techpreview-crds-$(1)
```

Have you tried setting the env var for the target?
```suggestion
update-codegen-techpreview-crds-$(1): OPENSHIFT_REQUIRED_FEATURESET=TechPreviewNoUpgrade
update-codegen-techpreview-crds-$(1): ensure-controller-gen ensure-yq ensure-yaml-patch
	OPENSHIFT_REQUIRED_FEATURESET=TechPreviewNoUpgrade $(call run-crd-gen,$(2),$(3))
.PHONY: update-codegen-techpreview-crds-$(1)
```
According to the [docs](https://www.gnu.org/software/make/manual/html_node/Target_002dspecific.html) this should set the env var for all of the dependencies of the target as well